mockfirestore/main.py

```diff
@@ -171,6 +176,41 @@ def limit(self, limit_amount: int) -> 'Query':
         self._limit = limit_amount
         return self
 
+    def start_at(self, document_fields: dict) -> 'Query':
+        self._start_at = (document_fields, True)
+        return self
+
+    def start_after(self, document_fields: dict) -> 'Query':
+        self._start_at = (document_fields, False)
+        return self
+
+    def end_at(self, document_fields: dict) -> 'Query':
+        self._end_at = (document_fields, True)
+        return self
+
+    def end_before(self, document_fields: dict) -> 'Query':
+        self._end_at = (document_fields, False)
+        return self
+    
+    def _cursor_helper(self, document_fields: dict, doc_snapshot: Iterator[List], before: bool, start: bool):
```

Comment:
I think `_apply_cursor` would be a more descriptive name.  Also it would be good to have the `Iterator[DocumentSnapshot]` return type hint.

Reply:
added hint as recommended.